src/Chapter4.hs

```diff
@@ -488,7 +494,36 @@ Implement the 'Applicative' instance for our 'List' type.
   may also need to implement a few useful helper functions for our List
   type.
 -}
-
+mapList :: (a -> b) -> List a -> List b
+mapList _ Empty = Empty
+mapList fun (Cons x tl) = Cons (fun x) (mapList fun tl)
+
+reverseList :: List a -> List a
+reverseList lst = go lst Empty
+  where
+    go :: List a -> List a -> List a
+    go Empty acc = acc
+    go (Cons hd tl) acc = go tl (Cons hd acc)
+
+concatList :: List a -> List a -> List a
+concatList lst1 lst2 = go (reverseList lst1) lst2
+  where
+    go :: List a -> List a -> List a
+    go Empty acc = acc
+    go (Cons hd tl) acc = go tl (Cons hd acc)
```

Comment:
This looks exactly like `reverse` 🤔 So you are doing reverse twice in here. I think this logic is a bit overcomplicated and can be done without `reverse` at all 🙂